Upstream changes and retrofit for python2

jsf9k opened the pull request:
I have no idea why the pulling in of upstream changes was squashed this time.

@climber-girl, can you try checking out this branch and running `vulnerability-age-data.py` via `cyhy-bash`?  I don't have `cyhy-bash` set up locally.

jsf9k committed:
Pull in upstream changes

jsf9k committed:
Retrofit source for python 2.7 for use with cyhy-bash

For the same reason, switch to using cyhy.db.database instead of
mongo_db_from_config.

jsf9k committed:
Remove now-unnecessary requirements from requirements.txt

dav3r reviewed (approved):
🐍 👍

mcdonnnj commented on vulnerability-age-data.py:
```diff
@@ -27,12 +26,10 @@
 import logging
 import sys
 
+from cyhy.db import database
```
If we're switching to `cyhy` for database functionality then I feel like the raw db access on lines 64, 71, and 90 should switch to the appropriate Doc classes to keep in line with other projects using that library.

jsf9k replied:
Addressed in 00a7d7b.

mcdonnnj reviewed (changes requested):
This looks solid, but please see my comment about switching to the `cyhy` library.

felddy reviewed (approved):
Looks good.  See @mcdonnnj change request.

jsf9k committed:
Use db.RequestDoc and db.TicketDoc instead of raw DB access

Also remove the get_all_descendents() function since it is no longer
needed.

Thanks to @mcdonnnj for the suggestion.

mcdonnnj reviewed (approved):
Thanks for changing that, LGTM. 👍

jsf9k committed:
Update README to match updated usage
